### PR TITLE
oss-fuzz hookup

### PR DESCRIPTION
**Pull request by keszybz**
A PR for oss-fuzz is also needed. I'll have that ready.

As mentioned in the commit message for 1/6, the error reporting when compressors are compiled-out is not very good. But I'd prefer to merge this as is, and work on the reporting later, because I think it's a major undertaking on its own. Right now, the error reporting is almost non-existent.

### CODE REVIEW

File: src/casync.c [hunk 1]
```diff
@@ -1485,6 +1485,8 @@ static int ca_sync_start(CaSync *s) {
                 if (r < 0)
                         return r;
 
+                (void) ca_store_set_compression_type(s->cache_store, s->compression_type);
```
Comment (poettering) [inline src/casync.c]: this should be unnecessary. Caches don#t recompress, they store everything as it is, i.e. compressed in the compressor of choice, or uncompressed depending on its current state. As such there's no need to propagate the compression type down to the cache store (but it doesn't hurt either)

**Comment by poettering** [comment]
CI failed, and it looks like it did so for a reason

**Comment by poettering** [comment]
looks good, but the cache store compression type propagation should be unnecessary, see above

File: src/casync.c (hunk 1 above)
Comment (keszybz) [reply src/casync.c]: OK. I don't quite understand this part of the code. The is that the cache was failing with ENOSYS for xz, when compiled without xz. If you have a nicer way to make the tests pass, I'm happy to change this.

Comment (poettering) [reply src/casync.c]: hmm, that indicates a bug I figure. Let's merge it as is then and I'll fix this up later.

**Pull request merged**